### PR TITLE
P1 features

### PR DESCRIPTION
**Pull request by Rohitprasad83**
- Added Search video using debouncing
- loaders
- No videos available screen.

wmW4epLNmK

### CODE REVIEW

**Review by ranitasahaa1230 — APPROVED** [review]
Got to learn something new
Clean UI, looks good to merge

File: src/App.css [hunk 1]
```diff
@@ -313,6 +312,53 @@
...
+    align-items: center;
+    justify-content: center;
+    gap: 2rem;
+}
+
+.add-videos>button {
+    width: max-content;
+    padding: 1rem 2rem;
+}
+
```
Comment (SanketDhabarde) [inline src/App.css:361]: Recommend to use media queries width in ems

**Review by SanketDhabarde — APPROVED** [review]
- Code structure looks good
- Good to merge

File: src/App.css (hunk 1 above)
Comment (Rohitprasad83) [reply src/App.css:361]: okay

**Pull request merged**